scripts/lint-style.lean

```diff
@@ -102,8 +102,53 @@ def allScriptsDocumented : IO Bool := do
     IO.println s!"error: found {undocumented.size} undocumented script(s): \
       please describe the script(s) in 'scripts/README.md'\n  \
       {String.intercalate "," undocumented.toList}"
-  return undocumented.size == 0
-
+  return undocumented.size
+
+/-- Verifies that all modules in `modules` are named in `UpperCamelCase`
+(except for explicitly discussed exceptions, which are hard-coded here).
+Return the number of modules violating this. -/
+def modulesNotUpperCamelCase (modules : Array Lean.Name) : IO Nat := do
+  -- Exceptions to this list should be discussed on zulip!
+  let exceptions := [
+    `Mathlib.Analysis.CStarAlgebra.lpSpace,
+    `Mathlib.Analysis.InnerProductSpace.l2Space,
+    `Mathlib.Analysis.Normed.Lp.lpSpace
+  ]
+  -- We allow only names in UpperCamelCase, possibly with a trailing underscore.
+  let badNames := modules.filter fun name ↦
+    let upperCamelName := Lake.toUpperCamelCase name
+    !exceptions.contains name &&
+    (upperCamelName != name && s!"{upperCamelName}_" != name.toString)
```

Comment:
I had some trouble parsing these lines, with the `let` and then recognizing this as the expression returned from the `fun` on line 118. Not sure how to fix it, but maybe indentation helps:
```suggestion
      (upperCamelName != name && s!"{upperCamelName}_" != name.toString)
```

Reply:
How do you like the current version?

Reply:
That works alright! (I suppose the idea is to split between checking for exceptions and checking for the name itself?)

Reply:
At the end of the day, you want to check "is my name in upperCamelCase, almost in upperCamelCase or one of the exceptions" --- and I don't care strongly about the order of these checks.